lib/msf/core/exe/segment_appender.rb

```diff
@@ -32,8 +32,13 @@ def generate_pe
 
       # Create a new section
       s = Metasm::PE::Section.new
-      s.name = '.' + Rex::Text.rand_text_alpha_lower(4)
-      s.encoded = payload_stub prefix
+			if @secname.blank?
+      	s.name = '.' + Rex::Text.rand_text_alpha_lower(4)
+      else
+				s.name = '.' + @secname
+				$stderr.puts "Created custom section \".#{secname}\""
+			end
+			s.encoded = payload_stub prefix
```

Comment:
For more readable and clean, please keep code consistently with others, we usually comply with the [ruby-style-guide](https://github.com/rubocop-hq/ruby-style-guide).